ansible.go

```diff
@@ -139,6 +141,52 @@ type AnsiblePlaybookRunner struct {
 	Env             []string // Envvars to pass into the Ansible run
 }
 
+func parseAnsibleRunStats(ansibleOutput AnsibleRunOutput) (AnsibleRunOutput, error) {
+
+	jsonErr := json.Unmarshal([]byte(ansibleOutput.CommandOutput.Stdout), &ansibleOutput)
+	if ansibleOutput.CommandOutput.Error != nil && jsonErr != nil {
+		logrus.Debug("Could not parse JSON from run. Ansible stdout:\n", ansibleOutput.CommandOutput.Stdout, "Ansible stderr:\n", ansibleOutput.CommandOutput.Stderr)
+	}
+	if ansibleOutput.CommandOutput.Error != nil {
+		return ansibleOutput, errors.Wrap(ansibleOutput.CommandOutput.Error, "ansible run failed")
+	}
+	if jsonErr != nil {
+		return parseDebugOutput(ansibleOutput)
+	}
+
+	return ansibleOutput, nil
+}
+
+func parseDebugOutput(ansibleOutput AnsibleRunOutput) (AnsibleRunOutput, error) {
+
+	lines := strings.Split(ansibleOutput.CommandOutput.Stdout, "\n")
+
+	recapRegex := regexp.MustCompile(`(\S+)\s+:\s+ok=(\d+)\s+changed=(\d+)\s+unreachable=(\d+)\s+failed=(\d+)\s+skipped=(\d+)\s+rescued=(\d+)\s+ignored=(\d+)`)
+
+	recapFound := false
+	for _, line := range lines {
+		if strings.Contains(line, "PLAY RECAP") {
+			recapFound = true
+			continue
+		}
+		if recapFound && strings.TrimSpace(line) != "" {
+			matches := recapRegex.FindStringSubmatch(line)
+
+			if len(matches) == 9 {
+				stats := fmt.Sprintf(`{"ok": %s, "changed": %s, "unreachable": %s, "failed": %s, "skipped": %s, "rescued": %s, "ignored": %s}`,
```

Comment:
This JSON string concat and parse seem unnecessary here. Also it seems that we are missing a layer of `{"stats": {"ok": xxx,...}}`.
Why can't we just set the values of `ansibleOutput.Stats` as a `AnsibleNodeStatus` struct? eg.

```
if ansibleOutput.Stats.Ok, err := strconv.Atoi(matches[2]); err != nil {
	return ansibleOutput, errors.Wrap(err, "failed to cast ok node count")
}
...
```